src/App.tsx

```diff
@@ -14,11 +15,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+enum Operation {
+  Add,
+  Remove,
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  goodsOperationsHandler = (operation: Operation, good: string) => {
+    switch (operation) {
+      case Operation.Add:
+        this.setState((state) => ({ selectedGoods: [...state.selectedGoods, good] }));
+
+        break;
+      case Operation.Remove:
+        this.setState((state) => (
+          { selectedGoods: state.selectedGoods.filter(goodItem => goodItem !== good) }
+        ));
+
+        break;
+      default:
+    }
+  };
```

Comment:
That makes the code much more readable to me, nice one!

Reply:
Thanks 😄